WordPress/Sniffs/WP/EnqueuedResourceParametersSniff.php

```diff
@@ -0,0 +1,155 @@
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+
+		// Check to see if a source ($src) is specified.
+		if ( ! isset( $parameters[2] ) ) {
+			return;
+		}
+
+		/*
+		 * Version Check: Check to make sure the version is set explicitly.
+		 */
+
+		if ( ! isset( $parameters[4] ) || 'null' === $parameters[4]['raw'] ) {
+			$this->phpcsFile->addError(
+				'Resource version not set in call to %s(). This means new versions of the resource will not always be loaded due to browser caching.',
+				$stackPtr,
+				'MissingVersion',
+				array( $matched_content )
+			);
+		} else {
+			// The version argument should have a non-false value.
+			if ( $this->is_falsy( $parameters[4]['start'], $parameters[4]['end'] ) ) {
+				$this->phpcsFile->addError(
+					'Version ($ver) is not explicitly set for %s; This means that the WordPress core version will be used, this is not recommend for Plugin or Theme development.',
+					$stackPtr,
+					'NoExplicitVersion',
+					array( $matched_content )
+				);
+			}
+		}
+
+		/*
+		 * In footer Check
+		 *
+		 * Check to make sure that $in_footer is set to true.
+		 * It will warn the user to make sure it is intended.
+		 *
+		 * Only wp_register_script and wp_enqueue_script need this check,
+		 * as it is not available to wp_register_style and wp_enqueue_style.
+		 */
+		if ( 'wp_register_script' !== $matched_content && 'wp_enqueue_script' !== $matched_content ) {
+			return;
+		}
+
+		if ( ! isset( $parameters[5] ) ) {
+			// If in footer is not set, throw a warning about the default.
+			$this->phpcsFile->addWarning(
+				'In footer ($in_footer) is set set explicitly %s; ' .
+				'It is recommended to load scripts in the footer, please set this value to `true`, or `false` if it should be loaded in the header.',
+				$stackPtr,
+				'NotInFooter',
+				array( $matched_content )
+			);
+		}
+	}
+
+	/**
+	 * Determine if a range has a falsy value.
+	 *
+	 * @param int $start The position to start looking from.
+	 * @param int $end   The position to stop looking (inclusive).
+	 *
+	 * @return bool True if the parameter is falsy.
+	 */
+	protected function is_falsy( $start, $end ) {
+		$falseTokens   = Tokens::$emptyTokens;
+		$falseTokens[] = T_FALSE;
+
+		// Find anything excluding the false tokens.
+		$hasNonFalse = $this->phpcsFile->findNext( $falseTokens, $start, $end + 1, true );
+		// If no non-false tokens are found, we are good.
+		if ( false === $hasNonFalse ) {
+			return true;
+		}
+
+		// Find out it has a number value.
+		$numberToken = $this->phpcsFile->findNext( array( T_DNUMBER, T_LNUMBER ), $start, $end + 1 );
+		if ( $numberToken ) {
+			$value = trim( $this->phpcsFile->getTokensAsString( $start, ( $end - $start ) + 1 ) );
+
+			// If there are characters that we don't want to eval, return not falsy.
+			if ( preg_match( '#^[\s(\d+(\.\d+)?)*/-]+$#', $value ) > 0 ) {
```

Comment:
Can any of these groups be marked as non-capturing (adding `?:` inside the `(`)?